### PR TITLE
[main] Issue #2301 - Sorting not honored with JPA and Oracle

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
Fixes Issue #2301

### CODE REVIEW

File: jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        public static void populate(Session session) {
+            List<Object> entities = Arrays.asList(ENTITY_FLOAT);
+            UnitOfWork unitOfWork = session.acquireUnitOfWork();
+            unitOfWork.removeAllReadOnlyClasses();
+            unitOfWork.registerAllObjects(entities);
+            unitOfWork.commit();
+        }
+
+        // Supported data types according to https://docs.oracle.com/cd/E19501-01/819-3659/gcmaz/
```
Comment (rfelcman) [inline jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java:128]: I don't see any reason why current pattern for table definitions in `org.eclipse.persistence.testing.models.jpa.advanced.AdvancedTableCreator#AdvancedTableCreator` should be avoided there.

File: jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+        public static EntityFloat[] ENTITY_FLOAT = new EntityFloat[] {
+                // Tallest and smallest length
+                new EntityFloat(70071, 17.0f, 17.1f, 7.7f, "testOLGH28289#70071"),
+                // Tallest and largest length
+                new EntityFloat(70077, 77.0f, 17.7f, 7.7f, "testOLGH28289#70077"),
+                new EntityFloat(70007, 70.0f, 10.7f, 0.7f, "testOLGH28289#70007")
+        };
+
+        public static void populate(Session session) {
```
Comment (rfelcman) [inline jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java:120]: This should be part of `setup` in org.eclipse.persistence.testing.tests.jpa.jpql.advanced.AdvancedQueryTest#testSetup like `EmployeePopulator`.

File: jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    @Override
+    public String toString() {
+        return String.format(
+                "EntityFloat{ id=%d, length=%f, width=%f, height=%f, description=\"%s\"}",
+                id, length, width, height, description);
+    }
+
+    // Based on reproduction scenario from issue #2301 (https://github.com/eclipse-ee4j/eclipselink/issues/2301)
+    public static final class Populator extends TogglingFastTableCreator {
```
Comment (rfelcman) [inline jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java:110]: I'd like prefer keep entity class simple as possible and not include data initialization there. It's usually `setUp` part of the test class.

File: jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java (hunk 1 above)
Comment (Tomas-Kraus) [reply jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java]: I won't change mi mind that it's better to have all those things on a single place than split into 3 differend modules. But I'll follow current pattern to keep code consistent.

File: jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java (hunk 3 above)
Comment (Tomas-Kraus) [reply jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java]: OK, moved to a separate populator.

File: jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java (hunk 2 above)
Comment (Tomas-Kraus) [reply jpa/eclipselink.jpa.testapps/jpa.test.advanced/src/main/java/org/eclipse/persistence/testing/models/jpa/advanced/entities/EntityFloat.java]: OK, moved to a separate populator.

**Review by rfelcman — APPROVED** [review]
LGTM

**Pull request merged**